### ISSUE TITLE
Memory leak in Menu API

### ISSUE THREAD

**Issue by ermenkov** [nwjs/nw.js#4860]
**Steps to reproduce:**
1. Have a simple app that uses the Menu API. Just package.json and index.html:
```html
<!DOCTYPE html>
<html lang="en">
<head>
	<meta charset="UTF-8">
    <script type="text/javascript">
        var menu = new nw.Menu();
        menu.append(new nw.MenuItem({ label: 'Item A' }));
    </script>
</head>
<body >
</body>
</html>
```
2. Start the app
3. Take a heap snapshot from the dev tools (*Profiling*)
4. Pick *Containment* from the drop down menu

**Observation 1**: 2 *Window* objects are present (Why not one but anyway)
![image](https://cloud.githubusercontent.com/assets/1094702/15318159/7d2bbe1c-1c2d-11e6-8380-ba61c480d140.png)

5. Make a *location.reload()* through the console multiple times
6. Take another heap snapshot

**Observation 2**: *Window* objects have multiplied by as many times you've reloaded the app
![image](https://cloud.githubusercontent.com/assets/1094702/15318182/9ca05c4e-1c2d-11e6-87ec-60ac842f6bc3.png)

**Expected**:
Since the *menu* variable is reinitialized on every load, the old one should be cleared by the GC 

*This bug is present in 0.12.3 and 0.14.5 and probably in all other versions in between*

**Comment by ermenkov** [comment]
This is a follow up on #4856

**Comment by ghostoy** [comment]
The issue is caused by a partial fix for #4313 (see aa3ed9157186dd0deaed6ebf07df792a20a845b8), which referenced context of AppWindow in the background page in order to preserve callbacks after navigation. I will revert that commit to prevent memory leak.

**Comment by rogerwang** [comment]
This is fixed in git and will be available in the next nightly build.
